**Using Technology: U.S. Economy Case Study**

U.S.. economic data 1976 to 1987

X1 = dollars/barrel crude oil

X2 = % interest on ten yr. U.S. treasury notes

X3 = foreign investments/billions of dollars

X4 = Dow Jones industrial average

X5 = GNP/billions of dollars

X6 = purchasing power U.S. dollar (1983 base)

X7 = consumer debt/billions of dollars

Reference: Statistical Abstract of the United States 103rd and 109th ediiton

In [ ]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [ ]:
df = pd.read_csv('tech_economy.csv')
df

CRUDE   INTEREST  FOREIGN         DJIA   GNP  PURCHASE    CONSUMER
0   10.900000   7.610000       31   974.900024  1718     1.757  234.399994
1   12.000000   7.420000       35   894.599976  1918     1.649  263.799988
2   12.500000   8.410000       42   820.200012  2164     1.532  308.299988
3   17.700001   9.439999       54   844.400024  2418     1.380  347.500000
4   28.100000  11.460000       83   891.400024  2732     1.215  349.399994
5   35.599998  13.910000      109   932.900024  3053     1.098  366.600006
6   31.799999  13.000000      125   884.400024  3166     1.035  381.100006
7   29.000000  11.110000      137  1190.300049  3406     1.000  430.399994
8   28.600000  12.440000      165  1178.500000  3772     0.961  511.799988
9   26.799999  10.620000      185  1328.199951  4015     0.928  592.400024
10  14.600000   7.680000      209  1792.800049  4240     0.913  646.099976
11  17.900000   8.380000      244  2276.000000  4527     0.880  685.500000

In [ ]:
df.describe

<bound method NDFrame.describe of         CRUDE   INTEREST  FOREIGN         DJIA   GNP  PURCHASE    CONSUMER
0   10.900000   7.610000       31   974.900024  1718     1.757  234.399994
1   12.000000   7.420000       35   894.599976  1918     1.649  263.799988
2   12.500000   8.410000       42   820.200012  2164     1.532  308.299988
3   17.700001   9.439999       54   844.400024  2418     1.380  347.500000
4   28.100000  11.460000       83   891.400024  2732     1.215  349.399994
5   35.599998  13.910000      109   932.900024  3053     1.098  366.600006
6   31.799999  13.000000      125   884.400024  3166     1.035  381.100006
7   29.000000  11.110000      137  1190.300049  3406     1.000  430.399994
8   28.600000  12.440000      165  1178.500000  3772     0.961  511.799988
9   26.799999  10.620000      185  1328.199951  4015     0.928  592.400024
10  14.600000   7.680000      209  1792.800049  4240     0.913  646.099976
11  17.900000   8.380000      244  2276.000000  4527     0.880  68

In [ ]:
df.shape

(12, 7)

In [ ]:
x = df.iloc[:, :-1].values
print(x)

[[1.08999996e+01 7.61000013e+00 3.10000000e+01 9.74900024e+02
  1.71800000e+03 1.75699997e+00]
 [1.20000000e+01 7.42000008e+00 3.50000000e+01 8.94599976e+02
  1.91800000e+03 1.64900005e+00]
 [1.25000000e+01 8.40999985e+00 4.20000000e+01 8.20200012e+02
  2.16400000e+03 1.53199995e+00]
 [1.77000008e+01 9.43999863e+00 5.40000000e+01 8.44400024e+02
  2.41800000e+03 1.37999999e+00]
 [2.81000004e+01 1.14600000e+01 8.30000000e+01 8.91400024e+02
  2.73200000e+03 1.21500003e+00]
 [3.55999985e+01 1.39099999e+01 1.09000000e+02 9.32900024e+02
  3.05300000e+03 1.09800005e+00]
 [3.17999992e+01 1.30000000e+01 1.25000000e+02 8.84400024e+02
  3.16600000e+03 1.03499997e+00]
 [2.90000000e+01 1.11099997e+01 1.37000000e+02 1.19030005e+03
  3.40600000e+03 1.00000000e+00]
 [2.86000004e+01 1.24399996e+01 1.65000000e+02 1.17850000e+03
  3.77200000e+03 9.61000025e-01]
 [2.67999992e+01 1.06199999e+01 1.85000000e+02 1.32819995e+03
  4.01500000e+03 9.27999973e-01]
 [1.46000004e+01 7.67999983e+00 2.09000000e+02 1.7

In [ ]:
y = df.iloc[:, -1].values
print(y)

[234.3999939 263.7999878 308.2999878 347.5       349.3999939 366.6000061
 381.1000061 430.3999939 511.7999878 592.4000244 646.0999756 685.5      ]


In [ ]:
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x,y)

LinearRegression()

In [ ]:
y_pred = regressor.predict(x)
np.set_printoptions(precision=2)
print(np.concatenate((y.reshape(len(y),1), y_pred.reshape(len(y_pred),1)) ,1))

[[234.4  232.09]
 [263.8  270.48]
 [308.3  313.57]
 [347.5  336.6 ]
 [349.4  341.29]
 [366.6  368.31]
 [381.1  373.88]
 [430.4  446.29]
 [511.8  522.64]
 [592.4  582.78]
 [646.1  647.46]
 [685.5  681.91]]


In [ ]:
print(regressor.predict([[20, 10, 300, 2500, 4700 ,0.80]]))

[620.51]


In [ ]:
print(regressor.coef_)

[-8.04e-01 -4.89e+00 -1.67e+00 -3.30e-03  3.85e-01  2.99e+02]


In [ ]:
print(regressor.intercept_)

-854.3870439556115


**Equation for the model is:**

consumer debt = -854.387 - 0.804(dollars/barrel) - 4.89(% interest on 10 yrs) -1.67(foreign investments/billions) - 0.0033 (Dow Jones industrial average) + 38.5 (GNP/billions of dollars) + 299 (purchasing power US dollar)

In [ ]:
from sklearn.metrics import mean_squared_error
mean_squared_error(y, y_pred)

66.23962709644108